flocker/control/test/test_persistence.py

```diff
@@ -249,3 +250,24 @@ def test_complex_keys(self):
                                manifestations={}, paths={},
                                devices={uuid4(): FilePath(b"/tmp")})
         self.assertEqual(node_state, wire_decode(wire_encode(node_state)))
+
+
+class ConfigurationMigrationTests(SynchronousTestCase):
+    """
+    Tests for ``_ConfigurationMigration``.
+    """
+    configurations_dir = FilePath(__file__).sibling('configurations')
+
+    def test_v1_v2_configuration(self):
+        """
+        A V1 JSON configuration blob is transformed to a V2 configuration
+        blob, with the result validating when loaded.
+        """
+        v1_json = self.configurations_dir.child(
+            b"configuration_v1.json").getContent()
+        v2_json = migrate_configuration(1, 2, v1_json)
+        v2_config = wire_decode(v2_json)
```

Comment:
Indeed, this is mostly a hangover from a previous design iteration where I had different Configuration parsing classes that determined the encoder and decoder methods used as the hooks in the wire methods. Then I decided that was unnecessarily complicated and reverted to keeping the wire methods as generic serialization functions, partly because they are also used in the AMP messaging service to encode/decode data chunks. But bear in mind these functions can operate on subsets of an overall config, so writing effective v1 to v2 tests will probably not be hard, we just won't be able to `wire_decode` an entire (non-latest) config in the real test implementations.